src/redux/reducers/main.ts

```diff
@@ -542,6 +543,58 @@ export const mainSlice = createSlice({
         previewConfigurationId: undefined,
       };
     },
+    updateResourceSubAction: (
+      state: Draft<AppState>,
+      action: PayloadAction<{
+        parentPayload: UpdateResourcePayload;
+        schemaVersion: string;
+        userDataDir: string;
+      }>
+    ) => {
+      const {schemaVersion, userDataDir} = action.payload;
+      const {isInClusterMode, resourceId, content, preventSelectionAndHighlightsUpdate} = action.payload.parentPayload;
+
+      try {
+        const currentResourceMap = isInClusterMode ? getLocalResourceMap(state) : getActiveResourceMap(state);
+        const resourceMap = state.resourceMap;
+        const resource = isInClusterMode ? resourceMap[resourceId] : currentResourceMap[resourceId];
+
+        const fileMap = state.fileMap;
+        if (resource) {
+          performResourceContentUpdate(resource, content, fileMap, resourceMap);
+          let resourceIds = findResourcesToReprocess(resource, currentResourceMap);
+          reprocessResources(
+            schemaVersion,
+            userDataDir,
+            resourceIds,
+            currentResourceMap,
+            fileMap,
+            state.resourceRefsProcessingOptions
+          );
+          if (!preventSelectionAndHighlightsUpdate) {
+            resource.isSelected = false;
+            updateSelectionAndHighlights(state, resource);
+          }
+        } else {
+          const r = resourceMap[resourceId];
+          // check if this was a kustomization resource updated during a kustomize preview
+          if (
+            r &&
+            (isKustomizationResource(r) || isKustomizationPatch(r)) &&
+            state.previewResourceId &&
+            isKustomizationResource(resourceMap[state.previewResourceId])
+          ) {
+            performResourceContentUpdate(r, content, fileMap, resourceMap);
+            processKustomizations(resourceMap, fileMap);
+          } else {
+            log.warn('Failed to find updated resource during preview', resourceId);
+          }
+        }
+      } catch (e) {
```

Comment:
what's the point of this catch if we're just throwing it again? the redux-logger will log the error!?